lib/services/access_request.go

```diff
@@ -1106,11 +1110,23 @@ func (m *RequestValidator) Validate(ctx context.Context, req types.AccessRequest
 		return trace.BadParameter("only promoted requests can set the promoted access list title")
 	}
 
+	// deduplicate requested resource IDs
+	var deduplicated []types.ResourceID
+	seen := make(map[string]struct{})
+	for _, resource := range req.GetRequestedResourceIDs() {
+		id := types.ResourceIDToString(resource)
+		if _, isDuplicate := seen[id]; isDuplicate {
+			continue
+		}
+		seen[id] = struct{}{}
+		deduplicated = append(deduplicated, resource)
+	}
+	req.SetRequestedResourceIDs(deduplicated)
```

Comment:
nit: it could be a minor optimization to only deduplicate if m.opts.expandVars is true, which is checked lower down in this function, and it will always get called before the request is created. There is also an argument against modifying a request after it is created so that the request being reviewed matches the backend state https://github.com/gravitational/teleport/blob/65c77a222e14bd0c4fb68bca4600ca54d4d5b808/lib/auth/auth.go#L5051-L5054

Reply:
What does `expandVars` have to do with request IDs though?

Reply:
Nothing in name, but in practice it's used to signal the request is being created and any modifications, normalizations, etc should happen, and those are skipped when the request is being read rather than written. We could rename it